### DISCUSSION TITLE
Aktuelle CSS Updates

### DISCUSSION THREAD

**Discussion by detLAN** [OpenCCU/OpenCCU#1506] [🇩🇪 Generelle Diskussionen]
Aktuell wird viel am CSS geschraubt, jedoch meiner Meinung nach in die falsche Richtung. Vieles wird komprimierter und der Platz bis an den Rand ausgenutzt. Modernes Webdesign geht genau in die andere Richtung, alles großzügiger und weiter auseinander.
Ein größerer Zeilenabstand und eine größere Schrift ist nicht nur besser zu erfassen, sondern sieht meiner Meinung nach auch besser aus. Es wirkt nicht so gedrungen wie es aktuell der Fall ist. Wenn ich z.B. die Startseite nehme und mir die Systemvariablen ansehe, da muss jetzt schon das "Streifenhörnchen" herhalten, damit es einigermaßen lesbar bleibt.
Die Angabe "Aktuelle Firmwareversion:" ist noch komischer, dieser Text wird umgebrochen, die Inhaltsangabe dann aber in der ersten Zeile wiedergegeben, anstatt hinter dem Doppelpunk. Btw. auf der Startseite wird von Firmware gesprochen, auf der Updateseite dann von Software - einheitlich ist dies nicht. Um bei der Startseite zu bleiben, ich würde die "Version" einfach weglassen und "Aktuelle Firmware" bzw. "Aktuelle Software" daraus machen. Dann muss der Beschreibungstext auch nicht umgebrochen werden und passt auf eine Zeile.

**Comment by Baxxy13** [comment]
Nightly von Heute?
Zeig doch mal wie es aussieht, habe gerade keinen Zugriff auf meine Testsysteme.
Im letzten Screenshot in #1495 wurde da eigentlich bei den fixen Informationen nichts umgebrochen.

**Comment by jens-maus** [comment]
> Aktuell wird viel am CSS geschraubt, jedoch meiner Meinung nach in die falsche Richtung. Vieles wird komprimierter und der Platz bis an den Rand ausgenutzt. Modernes Webdesign geht genau in die andere Richtung...

Na dann sag ich mal: Leg los! Setz dich hin und opfere mindestens genauso viel Zeit wie ich seit Jahren. Ich habe nun wirklich jahrelang genug darauf gewartet das sich mal nicht nur ein talentierter, sondern auch motivierter Web-Entwickler hier einfindet um vllt. ein WebUI 2.0 auf die Beine zu stellen.

Das ist aber nicht so eingetreten und so setze ich mich eben mit meinen eigenen Ideen und Fähigkeiten nun hin und ändere das um was ich kann und das **ich** selbst eben für sinnvoll erachte. Natürlich versuche ich hier+da auf konkrete Hinweise zu reagieren und Dinge auch einzuarbeiten. Aber einfach nur aus dem Wohnzimmer-Sessel heraus ein "geht in die falsche Richtung" herauszuposaunen ohne selbst konkret Hand anzulegen geht mir ehrlich gesagt nicht weit genug und auf einen solchen stylistischen Rundumschlag wie du das hier formulierst, kann ich auch nicht wirklich eingehen. Bring dich mit konkreten Verbesserungen ein, mach PullRequests auf oder fang an eine komplett neue WebUI zu entwickeln, dann haben wir eine Diskussionsbasis. Denn einfach nur allgemeine Kritik zu geben kann wirklich jeder - ist aber IMHO unterm Strich nicht wirklich hilfreich.

**Reply by detLAN** [reply]
> Bring dich mit konkreten Verbesserungen ein, mach PullRequests auf oder fang an eine komplett neue WebUI zu entwickeln, dann haben wir eine Diskussionsbasis.

Ich bin ja bereits konkret geworden, kann dies natürlich auch mit Bildern noch etwas verdeutlichen:
![SCR-20211106-j2g](https://user-images.githubusercontent.com/64675966/140610158-952d409b-49ca-4911-9652-d5378f0af537.png)
Hier die Darstellung der "Aktuelle Firmwareversion:", die Versionsnummer müsste eigentlich hinter dem Doppelpunkt dargestellt werden und nicht hinter "Aktuell". Und wie geschrieben, würde ich einfach das Wort "Version" weglassen, dann passt es auch optimal.

Gleich noch ein Bild zur Verdeutlichung:
![SCR-20211106-j1h](https://user-images.githubusercontent.com/64675966/140610248-0e92f602-e5f8-4b6b-8644-165fea723d58.png)
Auf der Startseite wird von "Firmware" gesprochen und hier steht dann überall "Software". Das könnte vereinheitlicht werden. Und wie geschrieben, erachte ich "Version" als überflüssig.
Auf dem Bild kann man auch sehr gut erkennen, dass z.B. die Beschreibungstexte in den Rahmen "eingequetscht" werden. etwas mehr Freiraum und ein etwas größerer Zeilenabstand wäre übersichtlicher. Dies gilt auch für die Schaltflächen, die könnten auch etwas großzügiger dargestellt werden.

Dies ist meine Meinung, nicht mehr und nicht weniger. Ich möchte damit niemandem zu nahe treten und habe dies nur zur allgemeinen Diskussion hier veröffentlicht.

**Reply by jens-maus** [reply]
Alles was du nun hier dargelegt hast ist eben "Geschmackssache" und teilweise auch davon abhängig was eQ3 uns bzgl. WebUI mit auf den Weg gibt. Solange wir nicht eine komplett eigene WebUI entwickeln/haben können wir nicht alles an allen stellen abändern weil wir sind Gefahr laufen für zukünftige OCCU/CCU3 Updates nicht mehr kompatibel zu sein. Das die WebUI style-technisch nicht dem aktuellen style-empfinden entspricht ist korrekt. Das lässt sich aber nur schwer eben abändern wenn man nicht komplett das ganze neu entwickeln möchte (was man eigentlich sollte). Und so muss man eben damit leben was wir hier **gemeinsam** hinbekommen. Hinweise/Meinungen/Geschmacksäußerungen können viele geben, konkrete Code-Änderungen eben nur wenige. Ich bin aber (wie man wissen sollte) für fast jede Schandtat zu haben und so auch prinzipiell für WebUI Anpassungen offen. Allerdings werde ich nicht auf reine Geschmacksäußerungen reagieren, sondern auf konkrete vorgeschlagene Code-Änderungen (am besten mittels PullRequests).

Wenn du also konkrete Vorschläge in Form von Code-Beispielen hast wie man an bestimmten Stellen in minimaler Form (am besten rein auf CSS Veränderungen) etwas verbessern kann, dann gerne her damit. Eins aber bitte nicht vergessen: Die CCU WebUI ist IMHO eine reine Konfigurationsoberfläche für das Einstellen/Konfigurieren der HomeMatic Geräte. Als Bedienoberfläche hat Sie IMHO schon längst ausgedient und sowas sollte in ioBroker, HomeAssistant oder ähnliches umgesetzt werden. Insofern lässt sich bereits darüber streiten inwieweit die WebUI von einer rein technischen Oberfläche mit 90er Jahre Style hin zu einer Oberfläche aus modernen Gesichtspunkten hin noch im bestehenden Framework geändert werden müsste/sollte.
